### ISSUE TITLE
Set Up Database

### ISSUE THREAD

**Issue by AhmadHRai** [Mosnad-Web01/Mosnad-3-Online-Learning-Platform#3]
Set up the MySQL database and create basic tables for core entities (users, courses, enrollments). Provide sample seed data to help with early testing and early integration with the frontend.

@feras-swed 
@Ahmed-Sitl

**Comment by HanaAlhemyari** [comment]
Note:  
Thank you for sending the database schema file via WhatsApp. It serves as a great description, and I’ve reviewed it. I’ve made the necessary adjustments and converted it into SQL statements.

I’ve only made a few minor modifications:

Removed the 'Students' and 'Instructors' tables and kept the 'Users' table with a specified role type.
Added the 'Lesson_Progress' table.
Updated the 'Reviews' and 'Courses' tables.
Added the 'User_Profile' table for storing user-specific data.
Everything looks good now. If you need any further adjustments or have questions, feel free to reach out.
[mysql.txt](https://github.com/user-attachments/files/17712719/mysql.txt)

**Comment by HanaAlhemyari** [comment]
I've added a simple role-based permission management system using MySQL. The system includes three primary roles:

Admin: Has all permissions, including managing courses, users, and viewing reports.
Instructor: Has medium-level permissions, such as creating and editing their own courses and managing their students.
Student: Has limited permissions for enrolling in and participating in courses through reviews.
To associate roles with permissions, I added a table for permissions, a table linking roles with permissions, and another table linking roles with users. 
This README.md provides a clear structure of roles, permissions, and practical use cases for each user role. The SQL commands for database structure and role-permission relationships are attached separately as requested.
[readme.md](https://github.com/user-attachments/files/17726562/readme.md)
